Objects/unicodeobject.c

```diff
@@ -6233,6 +6233,10 @@ int PyUnicode_Compare(PyObject *left,
     PyUnicodeObject *u = NULL, *v = NULL;
     int result;
 
+    if (!PyUnicode_Check(right) && PyErr_WarnPy3k_WithFix("comparing unicode and byte strings is not supported in 3.x", 
```

Comment:
I wonder if "convert the right string" is a bit too prescriptive -- potentially the correct fix is to convert the left string too. So maybe the warning should be more like "Comparing unicode and byte strings is not supported: ensure the strings in the comparison are either both unicode or both bytes"?